### PR TITLE
Clone repo #17

### LINKED ISSUE #17: Clone github repository in preparation for build
Required for #12 and #4. In order to know which repository to clone, we need #15.

### PR DESCRIPTION
**Pull request by ekorre1001**
fixes #17

### CODE REVIEW

File: src/main/java/ci/Clone.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    
+    // this function clones the repo and then checkout a specific commit
+    public static boolean gitClone(String repo_url, String local_dir, String commit_id) throws GitAPIException{
+
+            // Clone the repository into the local_directory
+            Git git = Git.cloneRepository().setURI(repo_url).setNoCheckout(true).setDirectory(new File(local_dir)).call();
+            // checkout commit with hash commit_id, detached HEAD state
+            git.checkout().setName(commit_id).call();
+
+            return true;
```
Comment (nolanderc) [inline src/main/java/ci/Clone.java:18]: Why does this function return a boolean?

File: src/main/java/ci/Clone.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.eclipse.jgit.api.errors.GitAPIException;
+import java.io.File;
+
+public class Clone {
+    
+    // this function clones the repo and then checkout a specific commit
+    public static boolean gitClone(String repo_url, String local_dir, String commit_id) throws GitAPIException{
+
+            // Clone the repository into the local_directory
+            Git git = Git.cloneRepository().setURI(repo_url).setNoCheckout(true).setDirectory(new File(local_dir)).call();
```
Comment (nolanderc) [inline src/main/java/ci/Clone.java:14]: How does this handle the case when `local_dir` already exists? Should probably be documented in the function comment.

File: src/main/java/ci/Clone.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package ci;
+
+import java.io.IOException;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import java.io.File;
+
+public class Clone {
```
Comment (nolanderc) [inline src/main/java/ci/Clone.java:8]: A more descriptive name for this class would probably be `Repository`, or similar.

**Review by nolanderc — CHANGES_REQUESTED** [review]

File: src/main/java/ci/Clone.java (hunk 1 above)
Comment (ekorre1001) [reply src/main/java/ci/Clone.java:18]: What kind of return type do we want? True means successfully cloned repo.

File: src/main/java/ci/Clone.java (hunk 3 above)
Comment (ekorre1001) [reply src/main/java/ci/Clone.java:8]: Good tip!

File: src/main/java/ci/Clone.java (hunk 1 above)
Comment (nolanderc) [reply src/main/java/ci/Clone.java:18]: If we only ever return `true`, the return type does not carry any meaning. Also, errors are treated as exceptions by the git library, so function completion == success, and throwing an exception == failure.

Comment (nolanderc) [reply src/main/java/ci/Clone.java:18]: ie. `void` is probably most appropriate

File: src/main/java/ci/Clone.java (hunk 2 above)
Comment (ekorre1001) [reply src/main/java/ci/Clone.java:14]: fixed!

File: src/main/java/ci/Clone.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package ci;
```
Comment (nolanderc) [inline src/main/java/ci/Clone.java:1]: The old file (`Clone.java`) should be removed

**Review by nolanderc — CHANGES_REQUESTED** [review]

**Review by nolanderc — APPROVED** [review]

**Pull request merged**